src/System.Net.HttpListener/src/System/Net/Managed/HttpEndPointListener.cs

```diff
@@ -56,7 +56,11 @@ public HttpEndPointListener(HttpListener listener, IPAddress addr, int port, boo
             {
                 _secure = secure;
                 // TODO #14691: Implement functionality to read SSL certificate.
+#if MONO
+                _cert = _listener.LoadCertificateAndKey (addr, port);
```

Comment:
Well, I would actually prefer to keep the Mono implementation as it is, but obsolete it - just so we're not breaking existing customer code.

And then we should work towards a general-purpose approach.

If you look at Mono's implementation https://github.com/mono/mono/blob/master/mcs/class/System/System.Net/HttpListener.Mono.cs#L69 - there are some problems with it.

It assumes to find a `<port>.cer` file for the certificate and a `<port>.pvk` file for the private key.  Then it uses Mono.Security's managed code to parse the private key and add it to the certificate.

There are two major problems with this approach - it doesn't support intermediate certificates and it does not interoperate very well with Mono's TLS providers (the native providers need to export certificate and key into binary format, then re-read it).  I would prefer something like using a PKCS#12 file and letting the native TLS provider read it.

It might also make sense to have a way of providing certificates on a per-application basis.